x-pack/legacy/plugins/security/server/lib/auth_redirect.js

```diff
@@ -15,6 +15,29 @@ import { wrapError } from './errors';
  * request that needs to be authenticated.
  */
 export function authenticateFactory(server) {
+
+  // Sometimes authentication subsystem may require some special headers to be sent back to
+  // the client with the requested payload (e.g. for mutual client-server authentication).
+  const authResponseHeadersMap = new WeakMap();
+  server.ext('onPreResponse', (request, h) => {
```

Comment:
Yeah, that's what I'm not completely sure about. We can either:

1. silently override headers if there any duplicates (current solution)
2. throw and fail if we detect duplicated headers
3. log with `warn`\`debug` all duplicated headers, but still override (current + log)

_Ideally_ authentication providers aren't supposed to abuse `authResponseHeaders` and should use them sparingly just for authentication specific headers (either defined by the spec or custom ES realms). Moreover plugins that register authentication providers (e.g. x-pack security or any NP plugin that depends on `security` in their manifest) should be considered as trusted since they'll have an exclusive access to `registerAuth` and friends in 7.x/master anyway and can potentially be very harmful in many ways.  

Having said that, I'm leaning towards 1 or 3, 2 sounds a bit crude to me. What do you think?

Reply:
I vote option 3 if it's not too much effort

Reply:
just for the records: we discussed this over Slack and agreed to keep it as is since it will go away in 7.4 (https://github.com/elastic/kibana/pull/41775) and checking for duplicates both boom errors and normal response makes this code looks too complex for the limited value it provides.